pkg/multitype/boolstring_test.go

```diff
@@ -0,0 +1,94 @@
+// Based on https://github.com/kubernetes/apimachinery/blob/455a99f/pkg/util/intstr/intstr.go
+
+package multitype
+
+import (
+	"testing"
+)
+
+func TestBoolOrString_Bool(t *testing.T) {
+	type fields struct {
+		Type    BoolOrStringType
+		BoolVal bool
+		StrVal  string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		want    bool
+		wantErr bool
+	}{
+		{
+			name:    "expect true when BoolVal is true",
+			fields:  fields{Type: Bool, BoolVal: true},
+			want:    true,
+			wantErr: false,
+		}, {
+			name:    "expect false when BoolVal is false",
+			fields:  fields{Type: Bool, BoolVal: false},
+			want:    false,
+			wantErr: false,
+		}, {
+			name:    "expect false when StrVal is 'true'",
+			fields:  fields{Type: String, StrVal: "true"},
+			want:    true,
+			wantErr: false,
+		}, {
+			name:    "expect false when StrVal is 'true'",
+			fields:  fields{Type: String, StrVal: "true"},
+			want:    true,
+			wantErr: false,
```

Comment:
These are duplicated, one should be "expect true when StrVal is 'true'"?

Reply:
nice catch, thanks